comp/forwarder/defaultforwarder/resolver/domain_resolver.go

```diff
@@ -210,19 +229,27 @@ func (r *SingleDomainResolver) GetAPIKeyVersion() int {
 // missing returns a list of elements that are in list a, but not in list b.
 // This is inefficient for large lists, but the assumption is that a config
 // will only have a very small number of API keys specified.
-// NOTE, this scrubs the API key to avoid leaking the key when logging.
 func missing(a []string, b []string) []string {
 	missing := []string{}
 
 	for _, key := range a {
 		if !slices.Contains(b, key) {
-			missing = append(missing, scrubber.HideKeyExceptLastFiveChars(key))
+			missing = append(missing, key)
 		}
 	}
 
 	return missing
 }
 
+// scrubKeys scrubs the API key to avoid leaking the key when logging.
+func scrubKeys(keys []string) []string {
```

Comment:
nit: no real need to create a new slice in this function, you could do it in-place. 

something along these lines:
```go
// scrubKeys scrubs the API key to avoid leaking the key when logging.
func scrubKeys(keys []string) []string {
	for i, key := range keys {
		keys[i] = scrubber.HideKeyExceptLastFiveChars(key)
	}
	return keys
}
```

I'm returning the `keys` slice only for good measure it's not even necessary since it can't grow. (I didn't check it compiles, just a suggestion).

Feel free to discard if you prefer working with a new slice for some reasons.